internal/service/advancedcluster/resource_advanced_cluster.go

```diff
@@ -895,6 +897,41 @@ func updateRequest(ctx context.Context, d *schema.ResourceData, projectID, clust
 	return cluster, nil
 }
 
+func syncAutoScalingConfigs(replicationSpecs *[]admin.ReplicationSpec20250101) {
+	if replicationSpecs == nil || len(*replicationSpecs) == 0 {
+		return
+	}
+
+	var defaultAnalyticsAutoScaling, defaultAutoScaling *admin.AdvancedAutoScalingSettings
+
+	for _, spec := range *replicationSpecs {
+		for i := range *spec.RegionConfigs {
+			regionConfig := &(*spec.RegionConfigs)[i]
+			if regionConfig.AutoScaling != nil && defaultAutoScaling == nil {
+				defaultAutoScaling = regionConfig.AutoScaling
+			}
+			if regionConfig.AnalyticsAutoScaling != nil && defaultAnalyticsAutoScaling == nil {
+				defaultAnalyticsAutoScaling = regionConfig.AnalyticsAutoScaling
+			}
+		}
+	}
+	applyDefaultAutoScaling(replicationSpecs, defaultAutoScaling, defaultAnalyticsAutoScaling)
+}
+
+func applyDefaultAutoScaling(replicationSpecs *[]admin.ReplicationSpec20250101, defaultAutoScaling, defaultAnalyticsAutoScaling *admin.AdvancedAutoScalingSettings) {
```

Comment:
Just to confirm, if a user effectively defines different auto scaling options in 2 separate regions the different values would be included in the patch and the user would face an expected error.

Reply:
Yes auto-scaling settings are supposed to be cluster-wide, so if they are different users will get an error OR if auto-scaling is configured for only selected regionConfigs, then also user will get an error. Will update the docs to reflect this.